packages/touchpoint-ui/src/types.ts

```diff
@@ -43,3 +43,18 @@ export interface ChoiceMessage {
    */
   messageIndex: number;
 }
+
+/**
+ * Custom Modalities allow rendering of rich components from nodes.
+ * See: https://docs.studio.nlx.ai/intentflows/documentation-flows/flows-build-mode/advanced-messaging-+-functionality#modalities
+ */
+export type CustomModalityComponent = FC<{
+  /**
+   * The name of the Modality as defined in Dialog Studio settings.
+   */
+  key: string;
+  /**
+   * The payload of the Custom Modality. The schema is defined in Dialog Studio settings.
+   */
+  data: any;
+}>;
```

Comment:
Might be nicer to do:

```suggestion
export type CustomModalityComponent<Data> = FC<{
  /**
   * The name of the Modality as defined in Dialog Studio settings.
   */
  key: string;
  /**
   * The payload of the Custom Modality. The schema is defined in Dialog Studio settings.
   */
  data: Data;
}>;
```

as this will allow the components to be strictly typed in typescript. 

I also wonder about having a prop called `key`, as this has a special meaning in React.

Reply:
I'll just remove the `key`, it's not used for anything.